datafusion/spark/src/function/datetime/date_add.rs

```diff
@@ -88,10 +90,12 @@ impl ScalarUDFImpl for SparkDateAdd {
 
 fn spark_date_add(args: &[ArrayRef]) -> Result<ArrayRef> {
     let [date_arg, days_arg] = args else {
-        return internal_err!(
-            "Spark `date_add` function requires 2 arguments, got {}",
-            args.len()
+        assert_eq_or_internal_err!(
```

Comment:
IMO this change does not improve it. We are already sure that the args length is not 2, so there is no need to check it again.
Also now this requires using `unreachable!()` at line 98